### ISSUE TITLE
When using 'initSelection' not calling 'select2-selecting' event

### ISSUE THREAD

**Issue by raphaelcarlosr** [select2/select2#1889]
When using 'initSelection' not calling 'select2-selecting' event
```js
initSelection: function(element, callback) {
    var id = element.val();
    if (id.isEmpty() == false) {
        $.ajax('{{URL_INIT_SELECTION}}', {
            type: 'POST',
            dataType: 'json',
            data: { param: '13456', id: 2}                
        }).done(function(result) { 
            var data = result.data.list[0];
            callback(data); 
            //i'm try this, but not work too o_O
            //element.trigger('select2-selecting', data); 
            //element.select2('val', data.Id); //its generate 'stackoverflow'
        });
    }
}

```

**Comment by kevin-brown** [comment]
Closing this off because of the lack of activity. If this is not fixed in 4.0, and you are interested in seeing it fixed, add a :+1: to the ticket to bring it to our attention.

If you can add additional details, that would also be useful.

**Comment by alex86sm** [comment]
`//on initselection
var evt = $.Event('select2:selecting', {params: data});
element.trigger(evt);
`

`//on event handler
on('select2:selecting', function(e, data){
     console.log(e.params);
})`
